### PR TITLE
Jira: CLOUDSTACK-8812 / CentOS 7

### PR DESCRIPTION
**Pull request by svenvogel**
shapeblue upstream 4.5.2 repository - corrected systemd service file

### CODE REVIEW

**Comment by remibergsma** [comment]
Hi @svenvogel Thanks! Can you please squash the 2 commits into 1 and also rebase with current master (make sure you have just one commit)? Then we'll be able to test it.

**Comment by bhaisaab** [comment]
Wait, I don't remember but did n't I merge this on master? I'm sure it is in 4.5. (sorry, in case I missed this fix).

**Comment by bhaisaab** [comment]
LGTM, please merge once tested. Thanks.

**Comment by remibergsma** [comment]
@svenvogel Something went wrong with the squashing as you now have 4 commits for this change. Can you make it a single commit please? Let me know if you need help.

**Comment by remibergsma** [comment]
@svenvogel Please update the title to something like this: "CLOUDSTACK-8812 Fix CentOS 7 systemd-tmpfiles - Operation not permitted" as it is more descriptive.

File: packaging/centos7/cloud-management.service [hunk 1]
```diff
@@ -15,22 +15,19 @@
...
 Description=CloudStack Management Server
 After=syslog.target network.target
 
 [Service]
 Type=forking
-ExecStart=/usr/sbin/tomcat-sysd start
-ExecStop=/usr/sbin/tomcat-sysd stop
+PermissionsStartOnly=true
+ExecStartPre=/usr/bin/systemd-tmpfiles --create
+ExecStart=/usr/sbin/cloudstack-management start
```
Comment (remibergsma) [inline packaging/centos7/cloud-management.service]: Why is this changed? In the RPM package this file is non existent so it fails.

```
-- Unit cloudstack-management.service has begun starting up.
Oct 10 18:41:09 cs1 systemd[3493]: Failed at step EXEC spawning /usr/sbin/cloudstack-management: No such file or directory
-- Subject: Process /usr/sbin/cloudstack-management could not be executed
-- Defined-By: systemd
```

**Comment by svenvogel** [comment]
Thirst

I don't know how I change it to one commit in github.

Second

I don't know why shapeblue change this for their version. I asked again which developer work on systemd. Maybe anyone have already an perfect startup script but I don't know. Shapeblue there are developer which work also on cloudstack. I am not a cloudstack dev. You know?

Third

Maybe centos 7 is supported in ‎4.6.

Fourth

I can change the name.

Mit freundlichen Grüßen

Sven Vogel

---------------------------------------------

Sven Vogel
Systems Engineer / Consultant
Kupper Computer GmbH

Prager Straße 15
04103 Leipzig


Registergericht Leipzig: HRB 19970

Geschäftsführer: Dipl.-Ing. Andreas Kupper

Tel:     +49 (0)341 305 36 0
Fax:    +49 (0)341 305 36 63
 Web:    http://www.kupper-computer.com
                 http://www.kupper-websolutions.de
Mail:    mailto:sven.vogel@kupper-computer.com
Von: Remi Bergsma
Gesendet: Samstag, 10. Oktober 2015 19:33
An: apache/cloudstack
Antwort an: apache/cloudstack
Cc: Vogel, Sven
Betreff: Re: [cloudstack] Jira: CLOUDSTACK-8812 / CentOS 7 (#888)



@svenvogel<https://github.com/svenvogel> Please update the title to something like this: "CLOUDSTACK-8812 Fix CentOS 7 systemd-tmpfiles - Operation not permitted" as it is more descriptive.

**Comment by remibergsma** [comment]
@svenvogel Thanks for getting back to me. As far as I know, there is no such thing as a ShapeBlue release. They provide packages which make it easy to install CloudStack. It is confusing though, pinging @bhaisaab to see if we can make it more generic like we discussed before.

Please remove the `ExecStart` change, as it doesn't work when I make packages from 4.6 branch:

```
cd packaging
./package.sh -d centos7
```
The result I then install and then it fails when starting `cloudstack-management`:

```
-- Unit cloudstack-management.service has begun starting up.
Oct 10 18:41:09 cs1 systemd[3493]: Failed at step EXEC spawning /usr/sbin/cloudstack-management: No such file or directory
-- Subject: Process /usr/sbin/cloudstack-management could not be executed
-- Defined-By: systemd
```

I will let you know how to make it one commit. Will do that when I'm back at my laptop.

CentOS 7 support is greatly improved in CloudStack 4.6 so this should work. If you want, I can see if I can take it from here and make it work as the `sudo` issue may be there and needs to be resolved.

File: packaging/centos7/cloud-management.service (hunk 1 above)
Comment (karuturi) [inline packaging/centos7/cloud-management.service]: I see a similar change in 4.5:packaging/centos63/rhel7/cloud-management.service

```
[Service]
Type=forking
ExecStartPre=/usr/bin/systemd-tmpfiles --create
ExecStart=/usr/sbin/cloudstack-management start
ExecStop=/usr/sbin/cloudstack-management stop
SuccessExitStatus=143
User=cloud
Group=cloud
```

**Pull request closed without merging**